packages/smart-contracts/src/lib/artifacts/SingleRequestProxyFactory/0.1.0.json

```diff
@@ -0,0 +1,284 @@
+{
+  "abi": [
+    {
+      "inputs": [
+        {
+          "internalType": "address",
+          "name": "_ethereumFeeProxy",
+          "type": "address"
+        },
+        {
+          "internalType": "address",
+          "name": "_erc20FeeProxy",
+          "type": "address"
+        },
+        {
+          "internalType": "address",
+          "name": "_owner",
+          "type": "address"
+        }
+      ],
+      "stateMutability": "nonpayable",
+      "type": "constructor"
+    },
+    {
+      "anonymous": false,
+      "inputs": [
+        {
+          "indexed": true,
+          "internalType": "address",
+          "name": "newERC20FeeProxy",
+          "type": "address"
+        }
+      ],
+      "name": "ERC20FeeProxyUpdated",
+      "type": "event"
+    },
+    {
+      "anonymous": false,
+      "inputs": [
+        {
+          "indexed": true,
+          "internalType": "address",
+          "name": "proxyAddress",
+          "type": "address"
+        },
+        {
+          "indexed": true,
+          "internalType": "address",
+          "name": "payee",
+          "type": "address"
+        },
+        {
+          "indexed": false,
+          "internalType": "address",
+          "name": "tokenAddress",
+          "type": "address"
+        },
+        {
+          "indexed": true,
+          "internalType": "bytes",
+          "name": "paymentReference",
+          "type": "bytes"
+        }
+      ],
+      "name": "ERC20SingleRequestProxyCreated",
+      "type": "event"
+    },
+    {
+      "anonymous": false,
+      "inputs": [
+        {
+          "indexed": true,
+          "internalType": "address",
+          "name": "newEthereumFeeProxy",
+          "type": "address"
+        }
+      ],
+      "name": "EthereumFeeProxyUpdated",
+      "type": "event"
+    },
+    {
+      "anonymous": false,
+      "inputs": [
+        {
+          "indexed": true,
+          "internalType": "address",
+          "name": "proxyAddress",
+          "type": "address"
+        },
+        {
+          "indexed": true,
+          "internalType": "address",
+          "name": "payee",
+          "type": "address"
+        },
+        {
+          "indexed": true,
+          "internalType": "bytes",
+          "name": "paymentReference",
+          "type": "bytes"
+        }
+      ],
+      "name": "EthereumSingleRequestProxyCreated",
+      "type": "event"
+    },
+    {
+      "anonymous": false,
+      "inputs": [
+        {
+          "indexed": true,
+          "internalType": "address",
+          "name": "previousOwner",
+          "type": "address"
+        },
+        {
+          "indexed": true,
+          "internalType": "address",
+          "name": "newOwner",
+          "type": "address"
+        }
+      ],
+      "name": "OwnershipTransferred",
+      "type": "event"
+    },
```

Comment:
I disagree. We should _not_ make tokenAddress indexed. We should _remove_ the `indexed` keyword from all event parameters except for `paymentReference`. 

The `indexed` keyword makes a given field easier to look up directly via an Ethereum Node RPC call. But we don't use Ethereum Nodes directly for indexing anymore. Instead, we run an off-chain indexer - the [payment-subgraph](https://github.com/RequestNetwork/payments-subgraph)

The `indexed` keyword costs a higher gas fee when emitting events. 

Since it is unused, we should remove it everywhere, except on the `paymentReference` - to align with the rest of the codebase.

This change will require a new deployment of the contract. We can replace the current 0.1.0 version instead of making a 0.2.0 version.

Reply:
There is no need to put it at 0.2.0 since we haven't officially released the current 0.1.0 version.